### DISCUSSION TITLE
To invoke additional cost function

### DISCUSSION THREAD

**Discussion by stefan-wallin-wns** [MilesCranmer/PySR#1117] [Q&A]
I want to add a consistency check on the generated expressions. The problem is that this test is complex and time consuming (seconds), hence cannot be used as the regular loss function. Is it possible to
1. Invoke such external routine at some stage in the training. E.g. for the most promising expressions?
2. Or, to regularly dump a set of promising expressions, do off-line checks, import the scores and continue searching?

**Comment by MilesCranmer** [comment]
Is it possible to build a fast surrogate model (approximation) of this external, expensive routine? That would be the easiest solution

**Comment by stefan-wallin-wns** [comment]
Thanks for the quick answer. No, it involves the use of the expression in a ODE solver. The data is local, the consistency check non-local.

**Comment by MilesCranmer** [comment]
Could you train a neural ODE first, and then fit a symbolic expression to the fitted model (as a neural net)?

Other than that I think the best approach is what you said - dump the expressions (you can use TensorBoardLogger for structured saving) and then manually evaluate them, and save the best ones.
